content/articles/2019-05-itdepends.Rmarkdown

```diff
@@ -0,0 +1,241 @@
+maintenance, lower functionality and introduce additional bugs.
+
+## dependencies are not equal
+
+One issue with striving for minimal dependencies is it treats all dependencies
+as equivalent. However in reality this is far from the case. There are many
+axis upon which dependencies can differ, but some of the most important include
+
+1. The number of additional dependencies. For example the
+   [rlang](https://github.com/r-lib/rlang) package has no dependencies apart
+   from R; while the [CNVscope](https://cran.r-project.org/package=CNVScope)
+   has 125 user dependencies, on both CRAN and
+   Bioconductor.
+
+2. The time taken to compile the package from source. Packages containing R code
+   generally install in roughly the same amount of time, but those that contain
+   C, C++ or Fortran code can take very different amounts of time depending
+   on the complexity of the code. For example the
+   [glue](https://cran.r-project.org/package=glue) package takes ~5 seconds to
+   compile on CRAN's build machines, whereas the
+   [readr](https://cran.r-project.org/package=readr) package takes ~500
+   seconds on the same machines.
+
+3. The size of the built binary package. Users installing binary packages need
+   to download them, so the size of the built binary is relevant, particularly
+   for those with slow internet connections. This also varies a great deal
+   across packages. The [awspack](https://cran.r-project.org/package=awspack)
+   binary is only 8.6 Kb in size, however the
+   [h2o](https://cran.r-project.org/package=h2o) package is 123 Mb, and the
+   largest binary on Bioconductor is over 4 Gb!
+
+4. The system requirements required to build and use the package. Some packages
+   require additional system dependencies in order to be used.
+   For instance the [rgdal](https://cran.r-project.org/package=rgdal) package
+   requires the GDAL and PROJ.4 dependencies, each of which in turn have
+   additional system dependencies. Another example is
+   [rJava](https://cran.r-project.org/package=rJava) which requires Java and
+   also has additional steps required to configure R for the proper Java
+   installation, which has caused [issues for many
+   people](https://stackoverflow.com/questions/tagged/rjava).
+
+I hope the above examples illustrate that not all package dependencies are
+equivalent and we should not treat them as such by using a simple count of
+dependencies as our metric.
+
+## prefer a balanced weight
+
+Instead of striving for absolutely minimal dependencies I would instead suggest
+a more holistic, balanced and quantitative approach.
+
+A holistic approach looks at the project as a whole and asks who is the primary
+audience? If the audience is other package authors than a leaner package with fewer
+dependencies may be more appropriate. If instead it is primarily data scientists or
+statisticians they will likely already have many popular dependencies installed and
+a more feature-full package is preferred.
```

Comment:
Also: consistency of features. I think of tidy eval and the name repair stuff as examples of us trying to make it increasingly impossible for different packages to offer NSE and name repair, but to implement them in slightly different ways by accident.